From: Milan Crha <mcrha@redhat.com>
Date: Tue, 26 Aug 2025 11:58:42 +0200
Subject: prefs: Fallback cursor size to 24 when invalid value in settings
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

When the settings contain an invalid value for the cursor size,
fallback the value internally to its default value (24), thus there
is shown a mouse cursor.

Suggested by Jonas Ådahl

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4592>

Bug-Ubuntu: https://bugs.launchpad.net/bugs/2125552
Origin: Upstream commit 870d7da6b137f5f69d243e95e1f715abe7baf75a
---
 src/core/prefs.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/core/prefs.c b/src/core/prefs.c
index 5f68f09..19369d8 100644
--- a/src/core/prefs.c
+++ b/src/core/prefs.c
@@ -1206,6 +1206,11 @@ meta_prefs_get_cursor_theme (void)
 int
 meta_prefs_get_cursor_size (void)
 {
+  if (cursor_size <= 0)
+    {
+      g_warning_once ("Invalid cursor size (%d) in settings, resetting to 24", cursor_size);
+      cursor_size = 24;
+    }
   return cursor_size;
 }
 
